mevislab.github.io/content/tutorials/basicmechanisms/macromodules.md

```diff
@@ -15,16 +15,108 @@ menu:
+Macro modules are primarily defined using the MeVisLab Definition Language (MDL) and often incorporate Python scripting for more dynamic behavior. You don't need to write C++ code to create a macro module.
+* **User Interface:**
+  You can define your own User Interface for macro modules using MDL.
+* **Scope:**
+Macro modules can be either global (available in all your MeVisLab projects) or local (specific to the current network you are working on).
+
+### Types of Macro modules: 
+#### Local Macro module:
+A Local Macro in MeVisLab is a macro module that exists only within the context of the current network document - i.e. it’s defined *locally* rather than being installed into the global module database. It does not require a package, it lives only inside the current network file (.mlab) you’re working on.
+
+*	A local macro is visible and editable only in the current network.
+*	A local macro is not listed in the Modules panel and module search.
+*	A local macro can only be reused elsewhere by copying it into another folder with your network file.
+
+#### Global Macro module:
+ A global macro module is stored in a central location within your MeVisLab installation's module directories (or a custom path) called Package. Once created, it appears in the module browser and can be used in any MeVisLab network you open. See [Package creation](/tutorials/basicmechanisms/macromodules/package/) for details about Packages.
+
+*	A global macro can be used in any MeVisLab network.
+*	A global macro is listed in the Modules panel and module search.
+   
+{{<alert class="info" caption="Info">}}
+Packages are the way MeVisLab organizes different development projects. You can organize your own modules, test cases or C++ modules in a package.
+{{</alert>}} 
+
+### Inputs, Outputs, and Fields:
+You can add inputs and outputs to your macro module to connect it with the other modules:
+* **Inputs:** These are connection points through which the module receives data from other modules. You define inputs in the *\*.script* file using the Input keyword. You can also define the type for each input.
+* **Outputs:** These are connection points through which the module sends processed data to other modules. You define outputs in the *\*.script* file using the Output keyword. You can also define the type for each output.
+
+You can add fields to your macro module. Fields allow you to change parameters for your module or to see the values of results. Fields can also be added to the panel of the macro module so that the user can change them.
+* **Fields:** These are parameters that control the module's behavior, typically visible in the module's panel or in the Module Inspector. You define fields in the *\*.script* file using the Field keyword, specifying the data type, default value, and other properties.
+The below figure shows the input, output and fields:
+![Inputs, Outputs, and Fields](/images/tutorials/basicmechanics/fields.png "Inputs, Outputs, and Fields")
+
```

Comment:
For fields, we can also mention that you have two options:
You can define your own fields providing your own parameters. Then you have to define the type of the field.

Alternative is to reuse a field of the internal network by providing the internalName of the internal field.